### PR TITLE
Clipper moved to public and extended by rect, filled triangle added

### PR DESCRIPTION
**Pull request by olliw42**
title says it

required for PR https://github.com/EdgeTX/edgetx/pull/131
needs to be merged before one can go on with that PR

:)

### CODE REVIEW

File: src/bitmapbuffer.cpp [hunk 1]
```diff
@@ -351,18 +351,18 @@ static float lb_min(float arr[], int n)
   return m;
 }
 
-bool BitmapBuffer::liangBarskyClipper(coord_t& x1, coord_t& y1, coord_t& x2, coord_t& y2)
+bool BitmapBuffer::liangBarskyClipper(coord_t & x1, coord_t & y1, coord_t & x2, coord_t & y2, coord_t xmin, coord_t xmax, coord_t ymin, coord_t ymax)
```
Comment (raphaelcoeffic) [inline src/bitmapbuffer.cpp:354]: Please avoid unnecessary changes like place of &.

File: src/bitmapbuffer.cpp [hunk 2]
```diff
@@ -351,18 +351,18 @@ static float lb_min(float arr[], int n)
...
   float p1 = -(x2 - x1);
   float p2 = -p1;
   float p3 = -(y2 - y1);
   float p4 = -p3;
 
-  float q1 = x1 - xmin;
-  float q2 = xmax - x1;
-  float q3 = y1 - ymin;
-  float q4 = ymax - y1;
+  float q1 = (x1 - xmin);
```
Comment (raphaelcoeffic) [inline src/bitmapbuffer.cpp:362]: This change is purely cosmetic, please avoid these.

**Review by raphaelcoeffic — CHANGES_REQUESTED** [review]

File: src/bitmapbuffer.cpp (hunk 1 above)
Comment (olliw42) [reply src/bitmapbuffer.cpp:354]: this is the style used in libopenui !

Comment (raphaelcoeffic) [reply src/bitmapbuffer.cpp:354]: The “style” is not normalized yet. It will be aligned to the clang-format included in EdgeTX.

Comment (olliw42) [reply src/bitmapbuffer.cpp:354]: I guess I am going to give up now ... too much things have added up which are too irrational to me ... I should better concentrate on doing my stuff. Many thx anyways.

Comment (raphaelcoeffic) [reply src/bitmapbuffer.cpp:354]: Does that mean I should finish this myself?

Comment (olliw42) [reply src/bitmapbuffer.cpp:354]: up to you guys, you also can close the PR. If you want me to I can close.

**Comment by raphaelcoeffic** [comment]
As @olliw42 refuse to discuss his changes, I'm closing this one.
